stdlib/public/core/Span/RawSpan.swift

```diff
@@ -286,18 +286,19 @@ extension RawSpan {
   ///   - span: An existing `Span<T>`, which will define both this
   ///           `RawSpan`'s lifetime and the memory it represents.
   @_alwaysEmitIntoClient
-  @lifetime(borrow span)
-  public init<Element: BitwiseCopyable>(
-    _elements span: borrowing Span<Element>
-  ) {
-    self.init(
-      _unchecked: span._pointer,
-      byteCount: span.count &* MemoryLayout<Element>.stride
+  @lifetime(span)
+  public init<Element: BitwiseCopyable>(_elements span: Span<Element>) {
+    let pointer = span._pointer
+    let rawSpan = RawSpan(
+      _unchecked: pointer,
+      byteCount: span.count == 1 ? MemoryLayout<Element>.size
+                 : span.count &* MemoryLayout<Element>.stride
```

Comment:
If this is correct then other APIs may need to be updated:

* `RawSpan.init(_unsafeStart:count:)` which uses `count * MemoryLayout<T>.stride`.

* `Span.withUnsafeBytes(_:)` which uses `_count * MemoryLayout<Element>.stride`.

* `Span.indices(of:)` which uses `stride &* other._count` and `stride &* _count`.

* `Span.init(_bytes:)` that consumes a `RawSpan`.

Alternatively, perhaps `CollectionOfOne` should be removed from the proposal?

Reply:
> The idea here is that safely converting a `Span` of 1 probably doesn't want to use `stride`; the need indeed comes from `CollectionOfOne`.

The same issue could affect nonempty `Span`s of any length:

```swift
let one: StaticString = ""
let two: (StaticString, StaticString) = ("", "")

withUnsafePointer(to: one) { start in
  let buffer = UnsafeBufferPointer(start: start, count: 1)
  return buffer.span.bytes.byteCount //-> either 17 or 24.
}

withUnsafePointer(to: two) { start in
  let start = start.pointer(to: \.0)
  let buffer = UnsafeBufferPointer(start: start, count: 2)
  return buffer.span.bytes.byteCount //-> either 41 or 48.
}
```

Should the padding after the *last* element always be excluded?

Reply:
Using tuples in this way is incorrect and always has been, because of that padding issue. The buffer in the `wUP(to: two)` closure is already invalid. We do get away with it with C imports because in that case `stride` is defined to be equal to `size`. We would really prefer to get this pattern (buffer-to-tuple) moved over to `InlineArray`.